web/src/SidebarResources.tsx

```diff
@@ -48,6 +49,72 @@ export function SidebarListSection(
   )
 }
 
+function SidebarItemsView(props: SidebarProps) {
+  return (
+    <>
+      {props.items.map((item) => (
+        <SidebarItemView
+          key={"sidebarItem-" + item.name}
+          item={item}
+          selected={props.selected === item.name}
+          pathBuilder={props.pathBuilder}
+          resourceView={props.resourceView}
+        />
+      ))}
+    </>
+  )
+}
+
+function SidebarLabelListSection(props: { label: string } & SidebarProps) {
+  if (props.items.length === 0) {
+    return null
+  }
+
+  return (
+    <>
+      <SidebarListSectionName>{props.label}</SidebarListSectionName>
+      <SidebarListSectionItems>
+        <SidebarItemsView {...props} />
+      </SidebarListSectionItems>
+    </>
+  )
+}
+
+function SidebarGroupedByLabels(props: SidebarProps) {
+  const labelsToResources: { [key: string]: SidebarItem[] } = {}
+  const unlabeled: SidebarItem[] = []
+
+  props.items.forEach((item) => {
+    if (item.labels.length) {
+      item.labels.forEach((label) => {
+        if (!labelsToResources.hasOwnProperty(label)) {
+          labelsToResources[label] = []
+        }
+
+        labelsToResources[label].push(item)
```

Comment:
i wonder if it's going to be weird to have an item in multiple groups

Reply:
fwiw it hasn't seemed odd to me while looking at all the designs. i think that only displaying a resource under the first label it has might be an unintuitive behavior, but it'll be interesting to get user feedback!